js/bybit.js

```diff
@@ -2751,43 +2885,26 @@ module.exports = class bybit extends Exchange {
             // 'stop_order_id': id, // only for conditional orders
             // 'p_r_trigger_price': 123.45, // new trigger price also known as stop_px
         };
-        let method = undefined;
-        if (market['swap']) {
-            if (market['linear']) {
-                method = 'privateLinearPostOrderReplace';
-            } else if (market['inverse']) {
-                method = 'v2PrivatePostOrderReplace';
-            }
-        } else if (market['future']) {
-            method = 'futuresPrivatePostOrderReplace';
-        }
-        const stopOrderId = this.safeString (params, 'stop_order_id');
-        if (stopOrderId !== undefined) {
-            if (market['swap']) {
-                if (market['linear']) {
-                    method = 'privateLinearPostStopOrderReplace';
-                } else if (market['inverse']) {
-                    method = 'v2PrivatePostStopOrderReplace';
-                }
-            } else if (market['future']) {
-                method = 'futuresPrivatePostStopOrderReplace';
-            }
-            request['stop_order_id'] = stopOrderId;
-            params = this.omit (params, [ 'stop_order_id' ]);
-        } else {
-            request['order_id'] = id;
-        }
+        const orderType = this.safeString (params, 'orderType');
+        const isStop = this.safeValue (params, 'stop', false);
+        const isConditionalOrder = isStop || (orderType === 'stop' || orderType === 'conditional');
+        params = this.omit (params, [ 'orderType', 'stop' ]);
+        const idKey = isConditionalOrder ? 'stop_order_id' : 'order_id';
+        request[idKey] = id;
         if (amount !== undefined) {
-            let qty = this.amountToPrecision (symbol, amount);
-            if (market['inverse']) {
-                qty = parseInt (qty);
-            } else {
-                qty = parseFloat (qty);
-            }
-            request['p_r_qty'] = qty;
+            request['p_r_qty'] = amount;
```

Comment:
Are we missing `exchange.amountToPrecision` here?

Reply:
Nop, it is done above, pls check this: https://github.com/ccxt/ccxt/pull/13263/files#diff-cb8b530c783df8bccc6e0df910e7433b34c9c9c3cea8904fcf2c59f1e60aeebeR2951

Reply:
I'd rather suggest that we do it in-place, otherwise we can't call createSpotOrder or createContractOrder, and I'm pretty sure the users will try calling those methods with the intellisense name-hinting in the IDE. Can we move those formatting lines inside the relevant methods?

Reply:
I'll move them but we should probably try to hide partial/helper methods from the users (in python we could do some name mangling or smth) because it is not guaranteed they will work standalone, users should not be calling random internal methods 😅

Reply:
They may look exchange-specific or arbitrary now, but we might have more of those in the future, and in general we should still behave unified as much as possible on all levels.

Reply:
@kroitor Nonetheless we will always have some partial/helper methods in order to make the code more reusable legible and maintainable we shouldn't be coding hard to read methods with hundreds of lines each